ios/brave-ios/Sources/Playlist/PlaylistManager.swift

```diff
@@ -557,34 +551,6 @@ public class PlaylistManager: NSObject {
     }
   }
 
-  private func deleteDanglingManagedAssets() {
-    if let playlistFolderPath = PlaylistDownloadManager.playlistDirectory {
-      do {
-        let urls = try FileManager.default.contentsOfDirectory(
-          at: playlistFolderPath,
-          includingPropertiesForKeys: nil,
-          options: [.skipsHiddenFiles]
-        )
-        for url in urls {
-          do {
-            // Playlist doesn't contain such an offline item, so it's dangling somehow and should be deleted.
-            if PlaylistItem.cachedItem(cacheURL: url) == nil {
-              try FileManager.default.removeItem(at: url)
```

Comment:
were these IO operations running on the main thread? Or does FileManager always run them on a separate thread? If not are there guards to ensure that we don't do file IO (or any other blocking operation) on the main thread?

Reply:
Yeah this code was not performant, we have to be more careful. Feel free to check the updated one here https://github.com/brave/brave-core/pull/24260/files a background thread is used